pyproject.toml

```diff
@@ -31,7 +31,9 @@ dependencies = [
     "aws-cdk.aws-lambda-python-alpha",
     "constructs>=10.0.0,<11.0.0",
     "boto3",
-    "opensearch-py"
+    "opensearch-py",
+    "requests",
+    "python-jose"
```

Comment:
These aren't CDK/infrastructure dependencies? I think they are Lambda dependencies, so should be in the `requirements.txt` file next to the lambda that needs them.

Reply:
True, I mainly wanted them installed so I could play around with them in the devcontainer if I was troubleshooting something for the lambdas (for example, import jose in a python terminal and make sure a certain function in that library worked).  I could either add these libraries to the devcontainer dockerfile, or just pip install them manually if I play around with them like that.

Reply:
Yeah, I'd prefer that. I like to keep **requirements** as minimal as possible. There is `optional.requires` where we added the `pip install .[dev]`, you could add some of those into another requires section and then install that way too?

Reply:
Putting the lambda dependencies in a different optional group will help keep them seperate, but still within the poetry dependency management. This can also be good for splitting up dev/prod/test dependencies. 

https://python-poetry.org/docs/master/managing-dependencies/#optional-groups